toolbox/library/src/YarpRpc.cpp

```diff
@@ -0,0 +1,257 @@
+            return false;
+        }
+    }
+
+    return blockInfo->parseParameters(m_parameters);
+}
+
+bool YarpRpc::configureSizeAndPorts(BlockInformation* blockInfo)
+{
+    // INPUT
+    // =====
+    //
+    // 1) Trigger signal
+    //
+    // OUTPUT
+    // ======
+    //
+    // No outputs
+    //
+
+    const bool ok = blockInfo->setPortsInfo(
+        {
+            // Inputs
+            {InputIndex::TriggerSignal, Port::Dimensions{1}, Port::DataType::DOUBLE},
+        },
+        {
+            // Outputs
+        });
+
+    if (!ok) {
+        bfError << "Failed to configure input / output ports.";
+        return false;
+    }
+
+    return true;
+}
+
+bool YarpRpc::initialize(BlockInformation* blockInfo)
+{
+    if (!Block::initialize(blockInfo)) {
+        return false;
+    }
+
+    // PARAMETERS
+    // ==========
+
+    if (!YarpRpc::parseParameters(blockInfo)) {
+        bfError << "Failed to parse parameters.";
+        return false;
+    }
+
+    // Temp variable to store rpc command string
+    std::string command;
+
+    bool ok = true;
+    ok = ok && m_parameters.getParameter("RisingEdgeTrigger", pImpl->risingEdgeTrigger);
+    ok = ok && m_parameters.getParameter("PortName", pImpl->serverPortName);
+    ok = ok && m_parameters.getParameter("RpcCommand", command);
+
+    if (!ok) {
+        bfError << "Failed to read input parameters.";
+        return false;
+    }
+
+    // CLASS INITIALIZATION
+    // ====================
+
+    pImpl->network = std::make_unique<yarp::os::Network>();
+
+    if (!yarp::os::Network::initialized() || !yarp::os::Network::checkNetwork(5.0)) {
+        bfError << "YARP server wasn't found active.";
+        return false;
+    }
+
+    // Open a rpc client port
+    pImpl->clientPortName = "...";
+
+    if (!pImpl->rpcClientPort.open(pImpl->clientPortName)) {
+        bfError << "Error while opening yarp rpc server port.";
+        return false;
+    }
+
+    if (!yarp::os::Network::connect(pImpl->rpcClientPort.getName(), pImpl->serverPortName)) {
+        bfError << "Failed to connect " << pImpl->serverPortName << " to "
+                << pImpl->rpcClientPort.getName() << ".";
+        return false;
+    }
+
+    // Store rpc command in bottle
+    pImpl->rpcCommand.fromString(command);
+
+    return true;
+}
+
+bool YarpRpc::terminate(const BlockInformation* /*blockInfo*/)
+{
+    if(pImpl->rpcClientPort.isWriting()) {
+        bfWarning << "Interrupting rpc port";
+        pImpl->rpcClientPort.interrupt();
+    }
+
+    yarp::os::Network::disconnect(pImpl->rpcClientPort.getName(), pImpl->serverPortName);
+
+    // Close the port
+    pImpl->rpcClientPort.close();
+
+    return true;
+}
+
+bool YarpRpc::output(const BlockInformation* blockInfo)
+{
+    InputSignalPtr triggerSignal = blockInfo->getInputPortSignal(InputIndex::TriggerSignal);
+
+    if (!triggerSignal) {
+        bfError << "Input signal not valid.";
+        return false;
+    }
+
+    bool rpcStatus = true;
```

Comment:
This is really a detail. I would initialize this to `false`. Not a big change in the current implementation, but it might hide possible errors if the cases below in the future will be modified and they contain a bug.

Reply:
@diegoferigo if it is set to false and if the trigger is never high then the output method always returns false. This is why I set the default value to be true and to be modified to false in case of calling the `sendRpcCommand()` method.

Reply:
You're right, I missed it.